pkg/api/secretgenerator/secretgenerator.go

```diff
@@ -0,0 +1,140 @@
+package secretgenerator
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/getlantern/deepcopy"
+
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/ci-tools/pkg/util/gzip"
+)
+
+func LoadConfigFromPath(path string) (Config, error) {
+	var config Config
+	cfgBytes, err := gzip.ReadFileMaybeGZIP(path)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := yaml.Unmarshal(cfgBytes, &config); err != nil {
+		return nil, err
+	}
+	return config.massiveItemGeneration()
+}
+
+type Config []SecretItem
+
+func (c Config) massiveItemGeneration() (Config, error) {
+	var processedItems []SecretItem
+	var errs []error
+	for _, item := range c {
+		items, err := item.generateItemsFromParams()
+		if err != nil {
+			errs = append(errs, err)
+		}
+		processedItems = append(processedItems, items...)
+	}
+	return processedItems, utilerrors.NewAggregate(errs)
+}
+
+func (c Config) itemsByName() map[string][]SecretItem {
+	byName := make(map[string][]SecretItem)
+
+	for _, secretItem := range c {
+		byName[secretItem.ItemName] = append(byName[secretItem.ItemName], secretItem)
+	}
+
+	return byName
+}
+
+func (c Config) IsItemGenerated(name string) bool {
+	byName := c.itemsByName()
+	if _, ok := byName[name]; ok {
+		return true
+	}
+	return false
+}
+
+func (c Config) IsComponentGenerated(name, component string) bool {
```

Comment:
Could we rename this to `IsFieldGenerated`? I know it conflicts with the bitwarden notion of field, but we don't use bitwarden anymore